pkgs/matcher/lib/src/core_matchers.dart

```diff
@@ -150,7 +150,8 @@ class _ReturnsNormally extends FeatureMatcher<Function> {
   @override
   bool typedMatches(Function f, Map matchState) {
     try {
-      f();
+      // ignore: unnecessary_cast
+      (f as Function)();
```

Comment:
can we do this?

```suggestion
      (f as void Function())();
```

Reply:
Unfortunately not, because it works today to do `expect(<T>(){}, returnsNormally);`, but a `<T>() -> void` is not a subtype of `() -> void`